pkg/registrar/registrar.go

```diff
@@ -102,6 +107,7 @@ func registerWorkloadController(mgr manager.Manager) error {
 		ConditionManagerBuilder: conditions.NewConditionManager,
 		ResourceRealizerBuilder: realizerworkload.NewResourceRealizerBuilder(repository.NewRepository, realizerclient.NewClientBuilder(mgr.GetConfig()), repository.NewCache(mgr.GetLogger().WithName("workload-stamping-repo-cache"))),
 		Realizer:                realizerworkload.NewRealizer(),
+		DependencyTracker:       dependency.NewDependencyTracker(2*defaultResyncTime, mgr.GetLogger().WithName("tracker-workload")),
```

Comment:
was there a particular reason for doing `2*defaultResyncTime` (emphasis on the **two**) which would mean 20h? I _imagine_ the reason is so that we guarantee that we don't let TTL expire before re-registering the intention to track

if I understood correctly (please correct me if I'm wrong), this essentially means that for anything that we want to track for enqueuing reconciliation requests, that currently being templates and serviceaccounts,

```
  template1 --> [workload1, ..., workloadN]
        ...
  template2 --> [workloadM, ..., workloadZ]

  sa1 --> [workloadK, ..., workloadP]
   ...
```

we'll keep enqueueing requests the workloads that got registered for those templates/serviceaccounts change for at least 20h, which is not a problem because we'd expect that templates and serviceaccounts not to be often changed, thus, keeping the simple strategy of letting the TTL expire being good enough (thus, not needing explicit removals from the tracker mapping)?

Reply:
Yes exactly. And it's not a problem if within those 20 hours something deleted gets enqueued as we'll bail out of that reconciler immediately. See #586 which will stop us from logging an error in this case